client/lib/fifo/fifo_test.go

```diff
@@ -45,7 +48,12 @@ func TestFIFO(t *testing.T) {
 	wait.Add(1)
 	go func() {
 		defer wait.Done()
-		io.Copy(&readBuf, reader)
+
+		reader, err = readerOpenFn()
+		require.NoError(err)
```

Comment:
Don't call t.Fatal in a non-main goroutine. Need to use a chan or `assert.NoError`